async-nats/src/jetstream/consumer/mod.rs

```diff
@@ -43,6 +46,70 @@ impl<T: IntoConsumerConfig> Consumer<T> {
         }
     }
 }
+impl<T: IntoConsumerConfig> Consumer<T> {
+    /// Retrieves `info` about [Consumer] from the server, updates the cached `info` inside
+    /// [Consumer] and returns it.
+    ///
+    /// # Examples:
+    ///
+    /// ```no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), async_nats::Error> {
+    /// use async_nats::jetstream::consumer::PullConsumer;
+    /// let client = async_nats::connect("localhost:4222").await?;
+    /// let jetstream = async_nats::jetstream::new(client);
+    ///
+    /// let mut consumer: PullConsumer = jetstream
+    ///     .get_stream("events").await?
+    ///     .get_consumer("pull").await?;
+    ///
+    /// let info = consumer.info().await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn info(&mut self) -> Result<consumer::Info, Error> {
```

Comment:
Return ref? Quite a big struct

```suggestion
    pub async fn info(&mut self) -> Result<consumer::Info, Error> {
```